src/rust/engine/src/context.rs

```diff
@@ -120,6 +105,12 @@ impl Context {
   }
 }
 
+impl StoreFileByDigest<Failure> for Context {
```

Comment:
I think `VFS` is implemented in `nodes.rs`... unclear what the pros/cons of splitting impls across files would be, beyond reducing cross-file imports, so *shrug.

Reply:
Moved